contracts/buy/PartyBuy.sol

```diff
@@ -0,0 +1,180 @@
+import {Party} from "../Party.sol";
+
+contract PartyBuy is Party {
+    // partyStatus Transitions:
+    //   (1) PartyStatus.ACTIVE on deploy
+    //   (2) PartyStatus.WON after successful buy()
+    //   (3) PartyStatus.LOST after successful fail()
+
+    // ============ Internal Constants ============
+
+    // PartyBuy version 1
+    uint16 public constant VERSION = 1;
+
+    // ============ Public Not-Mutated Storage ============
+
+    // the timestamp at which the Party is no longer active
+    uint256 public expiresAt;
+
+    // ============ Public Mutable Storage ============
+
+    // the maximum price that the party is willing to
+    // spend on the token
+    // NOTE: the party can spend *UP TO* 102.5% of maxPrice in total,
+    // and will not accept more contributions than this max amount
+    uint256 public maxPrice;
+
+    // ============ Events ============
+
+    // emitted when the token is successfully bought
+    event Bought(address triggeredBy, address targetAddress, uint256 ethPrice, uint256 ethFee, uint256 totalContributed);
+
+    // emitted if the Party fails to buy the token before expiresAt
+    // and someone closes the Party so folks can reclaim ETH
+    event Closed();
+
+    // ======== Constructor =========
+
+    constructor(
+        address _partyDAOMultisig,
+        address _tokenVaultFactory,
+        address _weth
+    ) Party(_partyDAOMultisig, _tokenVaultFactory, _weth) {}
+
+    // ======== Initializer =========
+
+    function initialize(
+        address _nftContract,
+        uint256 _tokenId,
+        uint256 _maxPrice,
+        uint256 _secondsToTimeout,
+        address _splitRecipient,
+        uint256 _splitBasisPoints,
+        string memory _name,
+        string memory _symbol
+    ) external initializer {
+        // initialize ReentrancyGuard and ERC721Holder
+        __ReentrancyGuard_init();
+        __ERC721Holder_init();
+        // set storage variables
+        nftContract = IERC721Metadata(_nftContract);
+        tokenId = _tokenId;
+        name = _name;
+        symbol = _symbol;
+        expiresAt = _secondsToTimeout + block.timestamp;
+        require(_maxPrice > 0, "PartyBuy::initialize: must set price higher than 0");
+        maxPrice = _maxPrice;
+        // validate that party split won't retain the total token supply
+        uint256 _remainingBasisPoints = 10000 - TOKEN_FEE_BASIS_POINTS;
+        require(_splitBasisPoints < _remainingBasisPoints, "PartyBuy::initialize: basis points can't take 100%");
+        // validate that a portion of the token supply is not being burned
+        if (_splitRecipient == address(0)) {
+            require(_splitBasisPoints == 0, "PartyBuy::initialize: can't send tokens to burn addr");
+        }
+        splitBasisPoints = _splitBasisPoints;
+        splitRecipient = _splitRecipient;
+        // validate token exists
+        require(_getOwner() != address(0), "PartyBuy::initialize: NFT getOwner failed");
```

Comment:
nit but it'd be nice to move all the requires to the top so we can save ppl gas if one of these require reverts

Reply:
if a transaction reverts, it eats up the entire gasPrice * gasLimit supplied to the tx, so there is no saving gas there. 

however, it may indeed be more readable to just have requires at the top, though. my thinking was "validate input -> set variable" and put the variables with no validation at the top. i will try this and see if it looks better :)